### DISCUSSION TITLE
filelock in multithreaded env

### DISCUSSION THREAD

**Discussion by saiaprameya** [tox-dev/filelock#329] [General]
Hi,
Does filelock work if i have multithreaded environment.

**Comment by kwist-sgr** [comment]
I guess this can help: [FileLocks and threads](https://py-filelock.readthedocs.io/en/latest/index.html#filelocks-and-threads)

**Reply by saiaprameya** [reply]
Thanks for the reply. But its re-entrant if thread_local flag is disabled. I dont want that to happen actually.
My use case is that i have multi process and each process has multiple threads. So for this kind of scenario it becomes very tough.
